### DISCUSSION TITLE
PrimeReact Calendar TimePicker - Close Dialog

### DISCUSSION THREAD

**Discussion by viktorjordan87** [primefaces/community#2874] [PrimeReact] [answered]
Hello,

I have with PrimeReact Calendar component when I am using with the time picker. Without timeformat, when I select a date the dialog close automatically, but with timepicker the dialog only close after select the date + time, if I click outside of the dialog. This is very annoying for the users. 

So my request is the following:
 a, If the time picker is active in the dialog bottom right corner there would be a close button
 b, The Button bar can hold another button "Close" if the time picker is active
 
 Question:
 c, Can I add somehow manually a button by myself to the button bar, so I can solve this problem?
 
 Thanks,
 Jordan

**Comment by qwadrox** [comment, marked as answer]
@viktorjordan87 hi!

You can add manually a button of course. There are a few available templates like the footer. There you can create your own "buttonbar". 

```ts
const calendarRef = useRef(null);

  const footerTamplete = () => {
    return (
      <div className="flex justify-content-between mt-2">
        <Button label="Today" onClick={setToToday} className="p-button-text" />
        <Button label="Clear" onClick={clearDate} className="p-button-text" />
        <Button
          label="Close"
          onClick={closeCalendar}
          className="p-button-text"
        />
      </div>
    );
  };

  const closeCalendar = () => {
    if (calendarRef.current) {
      calendarRef.current.hide();
    }
  };

  const setToToday = () => {
    setDateTime12h(new Date());
    closeCalendar();
  };

  const clearDate = () => {
    setDateTime12h(null);
    closeCalendar();
  };

  return (
    <div className="card flex flex-wrap gap-3 p-fluid">
      <div className="flex-auto">
        <label htmlFor="calendar-12h" className="font-bold block mb-2">
          12h Format
        </label>
        <Calendar
          id="calendar-12h"
          value={datetime12h}
          onChange={(e) => setDateTime12h(e.value)}
          showTime
          hourFormat="12"
          footerTemplate={footerTamplete}
          ref={calendarRef}
        />
      </div>
    </div>
  );
```

<img width="407" alt="image" src="https://github.com/user-attachments/assets/126f9ec6-62fa-445c-982a-20a8548c23c7">

try it: https://stackblitz.com/edit/b65qkb?file=src%2FApp.jsx

So you can programmatically clear, hide, and do [other sorts of things](https://primereact.org/calendar/#api.Calendar.methods) on the calendar.  Also, you could hide it when the [calendar changes](https://primereact.org/calendar/#api.Calendar.callbacks.onChange).

**Comment by viktorjordan87** [comment]
@qwadrox 
Thanks Máté

-----------------------------------------------------------------------------------------------------

Szia Máté, 

Köszönöm a megoldásod. Sokat segítettél. Be is építettem a komponensembe.

Üdvözlettel,
Viktor
